src/v1/feedback/db.ts

```diff
@@ -2,62 +2,170 @@ import { firebaseDB } from '../auth';
 import { Feedback } from './models/feedback'
 import { DISPLAY_MAP } from '../mapping';
 
+
 export class FeedbackDB {
 
-  // GET feedback of specific location
-  async feedbackListLocation(location: string):
-    Promise<{
-      id: string,
-      data: Feedback[]
-    }[]> {
-    const data = [];
-    if (location && location in DISPLAY_MAP) {
-      const docs = await firebaseDB.collection('feedback').doc(location).collection('feedback').get()
-        .then(doc => {
-          doc.forEach(d => {
-            data.push({ id: d.id, data: d.data() });
-          })
-        })
-        .catch(err => {
-          console.log(err);
-          return err;
-        });
-    }
-    return data;
+  // get hour docs
+  async getHourFeedback(dayRef: FirebaseFirestore.CollectionReference<FirebaseFirestore.DocumentData>, hour: string) {
+    const hourRef = await dayRef.doc(hour).collection('modelPrediction').get();
+    const predicts = hourRef.docs
+      .filter(doc => doc.id != "0")
+      .map(doc => ({ predictedWait: doc.id, data: doc.data() }));
+    return predicts;
+  }
+
+  // get day collections
+  async getDayFeedback(eateryRef: FirebaseFirestore.DocumentReference<FirebaseFirestore.DocumentData>, day: string) {
+
+    const dayRef = eateryRef.collection(day);
+    const hours = (await dayRef.get()).docs;
+    const data = await Promise.all(hours.map(async doc => {
+      const docID = doc.id;
+      const docData = await this.getHourFeedback(dayRef, docID);
+      return ({ hour: docID, data: docData })
+    }));
+    return data.filter(doc => doc.data.length > 0);
+
   }
 
-  // GET feedback of all locations
-  async feedbackList(feedbackListType) {
-    const data = [];
-    const loc = Object.keys(feedbackListType)
-      .map(location => this.feedbackListLocation(location)
-        .then(obj => {
-          if (obj.length !== 0) {
-            data.push({ eatery: location, size: obj.length, data: obj });
+  // get eatery docs
+  async getEateryFeedback(ref: FirebaseFirestore.CollectionReference<FirebaseFirestore.DocumentData>, eatery: string) {
+    const eateryRef = ref.doc(eatery);
+    const days = await eateryRef.listCollections();
+    const data = await Promise.all(days.map(async day => {
+      const dayID = day.id;
+      const dayData = await this.getDayFeedback(eateryRef, dayID);
+      return ({ day: dayID, data: dayData })
+    }));
+    return data.filter(doc => doc.data.length > 0);
+  }
+
+  // GET feedback
+
+  async getFeedback(
+    eatery?: string,
+    day?: string,
+    hour?: string,
+    predictedWait?: string,
+  ) {
+    const ref = firebaseDB.collection('feedbackData');
+
+    if (eatery) {
+      const eateryRef = ref.doc(eatery);
+
+      if (day) {
+        const dayRef = eateryRef.collection(day);
+        if (hour) {
+          // get data at specific wait time
+          const hourRef = dayRef.doc(hour);
+          if (predictedWait && predictedWait != "0") {
```

Comment:
Again, just wondering why predictedWait = 0 doesn't count. Maybe you could mention the reason in a comment?